tests/test_config.py

```diff
@@ -152,6 +152,19 @@ def test_from_pretrained_cache_dir_remote(self):
             PeftConfig.from_pretrained("ybelkada/test-st-lora", cache_dir=tmp_dirname)
             assert "models--ybelkada--test-st-lora" in os.listdir(tmp_dirname)
 
+    @parameterized.expand(ALL_CONFIG_CLASSES)
+    def test_save_pretrained_with_runtime_config(self, config_class):
```

Comment:
Let's also add a test that this works if users call `AutoModel.from_pretrained(..., ephemeral_transfer=True)`, and the same for `load_adapter` if the option is added there as well.

Reply:
I initially did, but then I realized that it doesn't actually support adding this flag in the first place.

In `from_pretrained` (inside `PeftModel`), it does this little magic:
```python
        hf_hub_download_kwargs, class_kwargs, _ = cls._split_kwargs(kwargs)
```
and this code requires that the kwarg (`ephemeral_transfers` in this case) is a property of the class. But it isn't. Because it's stored inside the `runtime` property. As such it's dropped in `_` and ignored.

Reply:
So, the `test_8bit_dora_cpu_offloading` test does check `PeftModel.from_pretrained` with `ephemeral_transfers=True`. I'm not sure what a good test would be that checks this beyond what is done already. Is there some feature that is not being tested properly in this scenario?

Reply:
In that test, you could save the model that has `ephemeral_transfer=True` at the end, then load the config and check that it's `False`.

Reply:
> Would it be cheating to enable ephemeral transfers in the LoraConfig in the `test_common_gpu/test_dora_ephemeral_transfers` starting model and then check that it's not there on load?

Yes, sorry, that's what I meant, not here.

Reply:
Got it! I think I figured out a good way to do that now. It's in the test_dora_ephemeral_transfers code. Specifically, it:
```
        peft_model = get_peft_model(model, config).eval()
        # Check that ephemeral transfers are present
        assert peft_model.peft_config['default'].runtime_config.ephemeral_transfers
```
then
```
            peft_model = PeftModel.from_pretrained(
                model,
                tmp_dir,
                device_map={"": "cpu"},
            ).eval()
            elapsed_time = time.perf_counter() - start_time

            # Check that ephemeral transfers are absent
            assert not peft_model.peft_config['default'].runtime_config.ephemeral_transfers
```